profile/benchmarking/simple.rb

```diff
@@ -0,0 +1,257 @@
+module Elasticsearch
+  module Benchmarking
+
+    # Class encapsulating all settings and functionality for running benchmarking
+    #   tests making simple requests.
+    #
+    # @since 7.0.0
+    class Simple
+
+      # Create a Simple request benchmark test.
+      #
+      # @example Create a test.
+      #   Benchmarking::Simple.new(:patron)
+      #
+      # @param [ Symbol ] adapter The adapter the client should be configured with.
+      #
+      # @since 7.0.0
+      def initialize(adapter = ::Faraday.default_adapter)
+        @adapter = adapter
+      end
+
+      # Run a Simple request benchmark test.
+      #
+      # @example Run a test.
+      #   Benchmarking::Simple.run(:ping)
+      #
+      # @param [ Symbol ] type The name of the test to run.
+      # @param [ Integer ] repetitions The number of test repetitions.
+      #
+      # @return [ Numeric ] The test results.
+      #
+      # @since 7.0.0
+      def run(type, opts={})
+        puts "#{type} : #{send(type, opts[:repetitions] || TEST_REPETITIONS, opts)}"
+      end
+
+      # Test sending a ping request.
+      #
+      # @example Test sending a ping request.
+      #   Benchmarking::Simple.ping(10)
+      #
+      # @param [ Integer ] repetitions The number of test repetitions.
+      #
+      # @since 7.0.0
+      def ping(repetitions, opts={})
+        results = repetitions.times.collect do
+          Benchmark.realtime do
+            1_000.times do
+              client.ping
+            end
+          end
+        end
+        Benchmarking.median(results)
```

Comment:
Thinking about this — it would be best to store stats for each iteration, but that would be obviously a _lot_ of documents. A good middle ground might be to return at least full stats (min/max/avg) from the run?